### PR TITLE
Magic Apiary Texture Rework + Fix Research

### PR DESCRIPTION
**Pull request by DrParadox7**
Retextures Apiary GUI to the same style as Forestry's Bee Houses with enhanced clarity for icons:

Increased Production:
image
Decreased Lifespan:
image
Increased Mutation:
image

Also fixed Research langs not correctly localized.

### CODE REVIEW

File: src/main/resources/assets/magicbees/textures/gui/apiary.png
Comment (Alastors) [inline src/main/resources/assets/magicbees/textures/gui/apiary.png:1]: No

**Review by Alastors — CHANGES_REQUESTED** [review]
Do not change the texture of the GUI

**Comment by Alastors** [comment]
The Magic Bees Apiary is a unique bit of thaumcraft, do not sully it by unifying textures when it is already so unique.

This PR as a whole is a reflection of someone who doesn't understand the vibrant uniqueness that each thaum addon brings to the table that is design.

Each texture, a kiss of specific style, a small impression of who the dev was, this is just sad.

File: src/main/resources/assets/magicbees/textures/gui/apiary.png
Comment (Dream-Master) [reply src/main/resources/assets/magicbees/textures/gui/apiary.png:1]: Maybe a mixture is nice. (gui saty icons changed) The new icons much cleaner expain where they stand for

Comment (Alastors) [reply src/main/resources/assets/magicbees/textures/gui/apiary.png:1]: But if he wants to put a proper side by side, he's free to do so, no PR that changes icons should be passed if they don't even at the very least do that

Comment (Dream-Master) [reply src/main/resources/assets/magicbees/textures/gui/apiary.png:1]: sure lets hear more people on this

Comment (Alastors) [reply src/main/resources/assets/magicbees/textures/gui/apiary.png:1]: apiary1
besides are you telling me this

makes less sense than this:
apiary2

They're both downright nonsensical, so it literally doesn't matter.

**Comment by Prometheus0000** [comment]
I agree with alastor, there's no reason to change the icons. *Maybe* the background, but I don't really think that needs changing either. If the author wanted it to use 'Forestry's Bee House icons' (which I don't recall ever seeing BTW) they would have used them.

**Comment by Alastors** [comment]
apiary
I'd like to bring to focus one final thing as well, base forestry doesn't have icons, so these icons are entirely unique to the Magic Apiary

**Comment by SilearFlare** [comment]
While you're at it, I'll pop in since the repo doesn't have any issues to actually start a discussion in and Alastor already did the same change between Gregapi and gregtech in https://github.com/GTNewHorizons/ThaumicHorizons/tree/gt6-gix, as mentioned in https://github.com/GTNewHorizons/ThaumicHorizons/issues/46, a specific commit in magic bees, specifically https://github.com/GTNewHorizons/MagicBees/commit/d9ef30fe32a71eb91bde9542c8fa52c4d2aa7ef1 crashes the modpack and using any version of magic bees alongside any version of gregtech outside of 5U causes a crash. 
At least in the case of thaumic horizons it was just a matter of enabling an external config which is disabled by default, in this case the mod just crashes with no option to do anything about it. and as it is, is impossible to play any version of magic bees past that commit.

Oh also the new icons are cool and I personally thing as an external observer that they should be changed :+1:

File: src/main/resources/assets/magicbees/textures/gui/apiary.png
Comment (bombcar) [reply src/main/resources/assets/magicbees/textures/gui/apiary.png:1]: Magic ain't technology no matter what some old sci-fi authors say. I say unique GUIs for the magical things is nice.

A gregtech style GUI should be for the beecompressor9000™ or some other technological disaster, not magic,

**Comment by DrParadox7** [comment]
These review are charming if not for their touch of creative flair.
MysteriousAges would appreciate a cleanup of Magic Bees' most cryptic aspect, especially after over a decade from its inception.

As a talented programmer artist yourself I get the apprehension, but when it comes to user interfaces, the ultimate aim should be making them feel cohesive and user-friendly which the original could use some improvement.

Anyways, you've asked for a comparison so here are the gifs (compression may slightly alter the colors).

**Old:**
Magic Apiary Animation-old

**New:**
Magic Apiary Animation-new

Things have been slightly tweaked in my latest iteration which is not included here but is available on request.

**Comment by Alastors** [comment]
> MysteriousAges would appreciate a cleanup of Magic Bees' most cryptic aspect, especially after over a decade from its inception.
Who is MysteriousAges?

> As a talented programmer artist yourself I get the apprehension, but when it comes to user interfaces, the ultimate aim should be making them feel cohesive and user-friendly which the original could use some improvement.
Ehhhhhhhhhh, not so much with Thaumcraft, the majority of Thaumcraft consistently and intentionally keeps the UI user-unfriendly and enigmatic for various reasons, and it's important to stay consistent with that design goal, though this system is one of the less enigmatic given that the Thaumonomicon directly lists out what each of them mean. Though I hardly believe a full rework of the GUI itself is tantamount to making it any more "user friendly" you could make the argument for the icons sure, but personally, I don't think there is enough value here to justify the utter destruction of the original GUI in totality.

**Comment by Alastors** [comment]
> > MysteriousAges would appreciate a cleanup of Magic Bees' most cryptic aspect, especially after over a decade from its inception.
> 
> ~Who is MysteriousAges?~ NVM you mean one of the devs, I fully invite him to come here and say the same then, though I would still disagree with that opinion personally. Of course that considered, he could just re-release the mod himself, given he still has ownership.
> 
> > As a talented programmer artist yourself I get the apprehension, but when it comes to user interfaces, the ultimate aim should be making them feel cohesive and user-friendly which the original could use some improvement.
> 
> Ehhhhhhhhhh, not so much with Thaumcraft, the majority of Thaumcraft consistently and intentionally keeps the UI user-unfriendly and enigmatic for various reasons, and it's important to stay consistent with that design goal, though this system is one of the less enigmatic given that the Thaumonomicon directly lists out what each of them mean. Though I hardly believe a full rework of the GUI itself is tantamount to making it any more "user friendly" you could make the argument for the icons sure, but personally, I don't think there is enough value here to justify the utter destruction of the original GUI in totality.
Screen Shot 2023-10-17 at 8 06 04 PM

> MysteriousAges would appreciate a cleanup of Magic Bees' most cryptic aspect, especially after over a decade from its inception.

Given there's been no update to Magic Bees in over 3 years though, I hardly believe you speak for MysteriousAges.

**Comment by Connor-Colenso** [comment]
Thanks for the rework, this second iteration looks much better imo.

**Comment by mitchej123** [comment]
I like the second iteration of the GUI. 


The icons seem slightly out of place for magic (at least the DNA one)

**Comment by Dream-Master** [comment]
i like to see it with bees ingame. How the bees look like on nearly black backround?

**Comment by DrParadox7** [comment]
> i like to see it with bees ingame. How the bees look like on nearly black backround?
Sure, I can do that:
in-use

**Comment by Dream-Master** [comment]
ok thanks for showing. I think the hexagons need to be more visible and also the slots.

**Comment by DrParadox7** [comment]
> The icons seem slightly out of place for magic (at least the DNA one)
I reluctantly agree with the sentiment. 
That DNA is definitely the weaker element of this design.

It is can be difficult to translate science elements such as genetic mutation into a magic theme.

I originally aimed to infuse a touch of alchemy, inspired by Fullmetal Alchemist/The Witcher 3 but I soon realized that this was quite a tough task, not just in the design-wise with a limited 10-pixel width canvas, but also in making sure it's easy to understand and use for everyone.

What you see is what I have settled on for lack of a better solution.

**Comment by Alastors** [comment]
After hours of stringent and fervent debate, we have now updated our dev Code of Conduct, and thusly, I push forth that this PR breaks GTNH CoC 7-1 given how this proposal would shift the identity significantly too far away from the unique feel intended by magic within the pack, likewise, the alternative modification proposed in #meta-dev can be handled by GTNH CoC 7-4

7) Regarding textures within GTNH
   1) Regarding magic textures: within the vision of the modpack, magic is meant to feel, look, and be different than GregTech, any form of GUI update for the sake of (or that would inadvertently bring) consistency with GT GUIs should be rejected.
   2) n/a
   3) n/a
   4) Regarding inventory changing: GUIs that shift the colour of the player's internal inventory are allowed as long as its usage is consistent. Examples being: GT's Steam Machines, Magic Bee's Magic Apiary, Railcraft's Coke Oven, etc.

**Comment by DrParadox7** [comment]
> ok thanks for showing. I think the hexagons need to be more visible and also the slots.

I covered your valid concern. 
There were also a couple of other issues fixed with it:
- Moved "Inventory"  label a tad bit to the left.
- The GUI was getting cut at the bottom by 2 pixel (also affects the original)
image
This can be tested from here:
https://github.com/DrParadox7/MagicBees/releases/tag/2.7.1-LE

**Comment by DrParadox7** [comment]
@Alastors If you've already altered the rules to predetermine the outcome of this PR, there's no need for further opinions. 
Enforcing your authority to conclude this discussion should be just as easy.

**Comment by SilearFlare** [comment]
Kinda defeats the whole point of having a discussion if ya can simply edit the rules to what can be done and can't be done mid-discussion, doesn't it? But then again I'm just here to point the issue with magic bees crashing with gt6 anyway :^)

**Comment by OneEyeMaker** [comment]
@DrParadox7 What about recolor/tint light grey part of UI to make it more unique? (with color of Greatwood/Arcane stone/Thaumium)

**Comment by bombcar** [comment]
This still looks like a GT machine to me - not really what I'd want, but I've not done bees or magic this far.

**Comment by Alastors** [comment]
> Kinda defeats the whole point of having a discussion if ya can simply edit the rules to what can be done and can't be done mid-discussion, doesn't it? But then again I'm just here to point the issue with magic bees crashing with gt6 anyway :^)

The new ruling came after tireless debate for 2 days straight, that's literally the point of having rules, make rules when new issues come up.

**Comment by Alastors** [comment]
> This still looks like a GT machine to me - not really what I'd want, but I've not done bees or magic this far.

This is the entire issue, it defaces Magic's identity for no real reason

**Comment by Alastors** [comment]
> @DrParadox7 New UI looks nice. But what about to recolor/tint light grey part of new UI to make it more unique? (with color of Greatwood/Arcane stone/Thaumium)

There was also concern in those debates (not even mostly by me) that the icons look too "techy" compared to what the original looked like

**Comment by boubou19** [comment]
what do we do with this PR?

**Comment by Alastors** [comment]
> what do we do with this PR?

It was rejected, and the CoC was changed to reflect it

**Comment by Alastors** [comment]
> We discuss it after stable

https://github.com/GTNewHorizons/GTNH-Dev-Doc/blob/master/developer's%20code%20of%20conduct.md 

Dream, respectfully we wasted 2 months on and off discussing this exact topic, please do not waste developer time further on such tripe like this. However, I will remind you of our policies regarding this topic now:

# Code of Conduct 
## Subsection 3: Content Creation
7. Regarding textures within GTNH
7-1. Regarding magic textures: within the vision of the modpack, magic is meant to feel, look, and be different than GregTech, any form of GUI update for the sake of (or that would inadvertently bring) consistency with GT GUIs should be rejected.
7-2. Regarding new textures: try to keep textures consistent with the mod/mod group they're being added to.
7-3. Regarding tooltips: reduction is fine when needed, but if there is a specific consistency, try to stay within that.
7-4. Regarding inventory changing: GUIs that shift the colour of the player's internal inventory are allowed as long as its usage is consistent. Examples being: GT's Steam Machines, Magic Bee's Magic Apiary, Railcraft's Coke Oven, etc.

I cannot stress this further, please do not waste any more of our time on meaningless debate on aesthetic taste, it gets incredibly argumentative, chaotic, downright rude, and quite frankly it's bad for developer moral on the pack. This has already been settled, we need to leave it in the past where it belongs.

Besides, even out of all the debate, no one actually cared for this specific UI, the most anyone actually cared about were the Icons or the rare desire to split the original UI instead.

**Comment by Connor-Colenso** [comment]
Stating this as a violation of the CoC makes no sense, the mod contains 3 GUIs, 2 of which are vanilla esque, see https://github.com/GTNewHorizons/MagicBees/blob/master/src/main/resources/assets/magicbees/textures/gui/jarScreen.png and https://github.com/GTNewHorizons/MagicBees/blob/master/src/main/resources/assets/magicbees/textures/gui/ringScreen.png and the third is this strange shaded variant.

So arguably this PR actually brings it fully in line with the CoC by giving the same shading to all GUIs?

**Comment by Caedis** [comment]
The color of the new GUI definitely feels too much like a GT/Vanilla one. I say we keep the old color of the gui, but apply that to all of Magic Bee's guis

**Comment by Alastors** [comment]
> Stating this as a violation of the CoC makes no sense, the mod contains 3 GUIs, 2 of which are vanilla esque, see https://github.com/GTNewHorizons/MagicBees/blob/master/src/main/resources/assets/magicbees/textures/gui/jarScreen.png and https://github.com/GTNewHorizons/MagicBees/blob/master/src/main/resources/assets/magicbees/textures/gui/ringScreen.png and the third is this strange shaded variant.
> 
> 
> 
> So arguably this PR actually brings it fully in line with the CoC by giving the same shading to all GUIs?

The bee ring and jar one are intentionally clones of each other, also the bee ring is not exactly in optimal shape, frankly I'm not sure if it actually has a recipe but using them as an example of two separate GUIs is very deceptive

**Comment by Steelux8** [comment]
The only problem I see here is that a DNA icon doesn't really fit magic content. Other than that, I look at the screenshots and what I think is Forestry bees with a touch of flair and a starry background that makes me think of Astral Sorcery, which *is* a magic mod. It overall looks fine to me, and I didn't really see it as a GT GUI the moment I looked at it.

**Comment by DrParadox7** [comment]
> The only problem I see here is that a DNA icon doesn't really fit magic content. Other than that, I look at the screenshots and what I think is Forestry bees with a touch of flair and a starry background that makes me think of Astral Sorcery, which _is_ a magic mod. It overall looks fine to me, and I didn't really see it as a GT GUI the moment I looked at it.

I agree.
This was the original concept for the DNA symbol but I had given up on it as it proved to be too difficult conceptualize it into pixel art (too tiny canvas size and awkward angles):
image
Feel free to have a go at it

**Comment by DrParadox7** [comment]
Made alterations to look less DNA-y and more runic (based on the Inguz rune shown above).
It should be satisfactory enough without compromising clarity.
image

**Comment by miaowwwwww** [comment]
tbh, i like the three new texture, how about just keep everything else, only update the 3 pics. i like the brown background and purple linings.

**Pull request closed without merging**